reactor-core/src/test/java/reactor/core/publisher/MonoCacheTimeTest.java

```diff
@@ -900,4 +905,57 @@ public void scanSubscriber() {
 		assertThat(test.scan(Scannable.Attr.CANCELLED)).isTrue();
 	}
 
+	@Test
+    public void signalDependentWithSchedulerOperatorTests() {
+	    Duration valueTtl = Duration.ofMillis(1000);
+        Duration errorTtl = Duration.ofMillis(2000);
+        Duration emptyTtl = Duration.ofMillis(3000);
+
+        AtomicInteger subCount = new AtomicInteger();
+        Supplier<Mono<Integer>> valueMonoSupplier = () -> Mono.just(subCount.incrementAndGet());
```

Comment:
I think the whole test with the `Supplier`, `Function` using a `Mono.defer(monoSupplier)`, etc... can be simplified:
 - turn these `Supplier`s into `Mono`s, but do pass each lambda to a `Mono.defer(lambda)` call (ie the lazyness of incrementing `subCount` comes from Mono.defer, not a Supplier)
 - make the `monoFunction` a `Function<Mono<Integer>, Mono<Integer>>`
 - remove the deferring from the function now that it is taken care of above

Reply:
Of course, that's a lot cleaner. Willdo!